### PR TITLE
Added messages.hpp and messages.cpp for message refactoring

### PR DESCRIPTION
**Pull request by efe-ozbatur**
## Description
Added messages.hpp and messages.cpp for message refactoring

## Changes
- messages.cpp doesn't have any code
- messages.hpp includes the messages and transitions.cpp uses the messages from that file

## Additional Info 
- stm variable is also available for easily writing "STM"

### CODE REVIEW

File: src/state_machine/messages.hpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+ */
+
+std::string stm_string = "STM";
+const char *stm = stm_string.c_str();
+
+std::string stop_command = "STOP command received";
+const char *m_stop = stop_command.c_str();
+
+std::string critical_failure_navigation = "Critical failure in navigation";
+const char *m_critical_navigation = critical_failure_navigation.c_str();
```
Comment (mifrandir) [inline src/state_machine/messages.hpp]: Only the declaration, i.e. 
```
std::string stop_command;
```
belongs in the header. The actual literal should be in the .cpp file. 

This way the string literal gets redefined every time a .cpp file (implicitly) includes the messages.hpp header and thus the linker cannot be sure which memory location is actually being referred to.

**Comment by mifrandir** [comment]
Also, the duplication between `std::string` and the C-style strings `char *` seems a bit clunky here. Do we really need the latter? If the logger doesn't support this (which I'd be surprised by), it shouldn't be too difficult to add this feature.

**Comment by mifrandir** [comment]
This is required because we're stuck on C++11 and inline variables are only available from C++17 onwards...

It ain't too pretty, but I think it's fine.

**Review by mifrandir — APPROVED** [review]
It looks fine to me for now, though this is not yet as convenient as I'd llike it to be. (But it will do for testing, so this is fine.)

**Review by DonaldJennings — APPROVED** [review]
Very nice refactoring 👍🏻 Looks good to me :).

**Review by mifrandir — CHANGES_REQUESTED** [review]
You are missing [this](https://github.com/Hyp-ed/hyped-2021/blob/9b15a590de4e54ab5c4474ddec22f96f37092000/src/state_machine/state.hpp#L81).

**Comment by mifrandir** [comment]
TL;DR: Don't import what you don't use!

**Review by mifrandir — APPROVED** [review]
